src/java.base/share/classes/javax/net/ssl/SSLEngine.java

```diff
@@ -336,6 +336,25 @@
  * started, an {@code SSLEngine} can not switch between client and server
  * modes, even when performing renegotiations.
  * <P>
+ * The ApplicationProtocol {@code String} values returned by the methods in
+ * this class are in the network byte representation sent by the peer and
+ * may need to be converted to its Unicode format before use.  For example,
+ * if an ALPN value was encoded using {@code UTF-8}, the {@code String}
+ * should be converted to its {@code byte[]} representation and then
+ * {@code UTF-8}-decoded to a {@code String} before use.
+ *
+ * <blockquote><pre>
+ *     String networkString = sslEngine.getHandshakeApplicationProtocol();
+ *     String unicodeString = new String(networkString.getBytes(
+ *             StandardCharsets.ISO_8859_1), StandardCharsets.UTF_8);
+ *
+ *     // MEETEI MAYEK LETTERS HUK UN I (Unicode 0xabcd->0xabcf)
+ *     if (unicodeString.equals("\uabcd\uabce\uabcf") {
+ *         ...
```

Comment:
Hi Brad,

There's a missing closing parenthesis here on line 352. 

Additionally - the unicode characters in the string above will be substituted by the compiler before the API document is generated. I am suspecting that this is not what you want. If you want to see the literal unicode escape in the generated documentation, you will need to employ some tricks. One of them could be to use the unicode escape of \ instead of \ to prevent the compiler from interpreting \uabcd as a unicode escape.

Something like:

```
 *     // MEETEI MAYEK LETTERS HUK UN I (Unicode 0xabcd->0xabcf)
 *     if (unicodeString.equals("\u005cuabcd\u005cuabce\u005cuabcf")) {
```

would do the trick. Alternatively - this would probably work too:

```
 *     // MEETEI MAYEK LETTERS HUK UN I (Unicode 0xabcd->0xabcf)
 *     {@code if (unicodeString.equals("\}{@code uabcd\}{@code uabce\}{@code uabcf"))} {
```

I realize none of these alternatives are ideal - maybe someone knows a better trick...

Reply:
I made this change in SSLParameters, and forgot that I had a similar change to make in SSLEngine/SSLSocket.